xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/java/org/xwiki/extension/security/internal/ExtensionSecurityJob.java

```diff
@@ -0,0 +1,109 @@
+import org.xwiki.extension.index.internal.security.ExtensionAnalysisResult;
+import org.xwiki.extension.repository.InstalledExtensionRepository;
+import org.xwiki.extension.security.ExtensionSecurityEvent;
+import org.xwiki.extension.security.analyzer.ExtensionSecurityAnalyzer;
+import org.xwiki.extension.security.internal.analyzer.VulnerabilityIndexer;
+import org.xwiki.extension.security.internal.analyzer.osv.OsvExtensionSecurityAnalyzer;
+import org.xwiki.job.AbstractJob;
+import org.xwiki.job.DefaultJobStatus;
+
+import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage;
+
+/**
+ * Run a security analysis on the current instance's extensions.
+ *
+ * @version $Id$
+ * @since 15.5RC1
+ */
+@Component
+@InstantiationStrategy(ComponentInstantiationStrategy.PER_LOOKUP)
+@Named(ExtensionSecurityJob.JOBTYPE)
+public class ExtensionSecurityJob
+    extends AbstractJob<ExtensionSecurityRequest, DefaultJobStatus<ExtensionSecurityRequest>>
+{
+    /**
+     * The id of the job.
+     */
+    public static final String JOBTYPE = "extension_security";
+
+    @Inject
+    private InstalledExtensionRepository installedExtensionRepository;
+
+    @Inject
+    @Named(OsvExtensionSecurityAnalyzer.ID)
+    private ExtensionSecurityAnalyzer extensionSecurityAnalyzer;
+
+    @Inject
+    private VulnerabilityIndexer vulnerabilityIndexer;
+
+    @Override
+    public String getType()
+    {
+        return JOBTYPE;
+    }
+
+    @Override
+    protected void runInternal()
+    {
+        Collection<InstalledExtension> installedExtensions =
+            this.installedExtensionRepository.getInstalledExtensions();
+        this.progressManager.pushLevelProgress(installedExtensions.size(), this);
+
+        try {
+            // Note: for now, this step is sequential and each extension is analyzed after the previous one.
+            long newVulnerabilityCount = 0;
+            for (InstalledExtension extension : installedExtensions) {
+                this.progressManager.startStep(this);
+                try {
+                    ExtensionAnalysisResult analysis = this.extensionSecurityAnalyzer.analyze(extension);
+                    if (analysis != null) {
+                        boolean update = this.vulnerabilityIndexer.update(extension, analysis);
+                        if (update) {
+                            newVulnerabilityCount++;
+                        }
+                    }
+                } catch (ExtensionSecurityException e) {
+                    this.logger.warn("Failed to analyse [{}]. Cause: [{}]", extension, getRootCauseMessage(e));
+                } catch (Exception e) {
+                    this.logger.warn("Unexpected error [{}]", getRootCauseMessage(e));
+                }
+                this.progressManager.endStep(this);
+            }
+            if (newVulnerabilityCount > 0) {
+                this.observationManager.notify(new ExtensionSecurityEvent(), this, newVulnerabilityCount);
```

Comment:
I don't understand why the source of the event is `this`? I mean you cannot really exploit this information... Feels like the event should be triggered for each new update of the security indexer no?